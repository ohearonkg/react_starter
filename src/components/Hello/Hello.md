```
<Hello name={"Sarah"} age={12} />
```
