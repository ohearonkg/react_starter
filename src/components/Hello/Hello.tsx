import * as React from "react";
import styled from "react-emotion";

export interface IHello {
  /**
   * The name of person
   * to be printed
   */
  name: string;

  /**
   * The age of the person
   * to be printed
   */
  age: number;
}

const HelloWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex-direction: column;
`;

const ItemWrapper = styled.div`
  flex: 1 1 auto;
  text-align: center;
`;

const NameHeading = styled.h1`
  margin: 0;
  font-size: 32px;
`;

const AgeHeading = styled.h2`
  margin: 0;
  font-size: 24px;
`;

const Hello = (props: IHello) => {
  return (
    <HelloWrapper>
      <ItemWrapper>
        <NameHeading> Hello {props.name} </NameHeading>
      </ItemWrapper>

      <ItemWrapper>
        <AgeHeading> Age {props.age} </AgeHeading>
      </ItemWrapper>
    </HelloWrapper>
  );
};

export default Hello;
