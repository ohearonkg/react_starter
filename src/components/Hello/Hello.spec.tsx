import * as emotion from "emotion";
import { mount, ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import { createSerializer } from "jest-emotion";
import * as React from "react";
import Hello from "./Hello";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("Hello Component", () => {
  /**
   * We create a new shallowly
   * mounted instance of our component
   * before each test
   */
  let wrapper: ShallowWrapper;
  let mounted: ReactWrapper;

  beforeEach(() => {
    wrapper = shallow(<Hello name="Kevin" age={12} />);
    mounted = mount(<Hello name="Kevin" age={12} />);
  });

  it("Should match its snapshot", () => {
    expect(toJson(mounted)).toMatchSnapshot();
  });

  it("Should render the value passed to its name prop", () => {
    expect(wrapper.html()).toMatch("Kevin");
  });
});
