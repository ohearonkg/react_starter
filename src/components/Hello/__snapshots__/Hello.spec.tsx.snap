// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component Should match its snapshot 1`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  text-align: center;
}

.emotion-0 {
  margin: 0;
  font-size: 32px;
}

.emotion-4 {
  margin: 0;
  font-size: 24px;
}

<Hello
  age={12}
  name="Kevin"
>
  <HelloWrapper>
    <div
      className="emotion-8 emotion-9"
    >
      <ItemWrapper>
        <div
          className="emotion-2 emotion-3"
        >
          <NameHeading>
            <h1
              className="emotion-0 emotion-1"
            >
               Hello 
              Kevin
               
            </h1>
          </NameHeading>
        </div>
      </ItemWrapper>
      <ItemWrapper>
        <div
          className="emotion-2 emotion-3"
        >
          <AgeHeading>
            <h2
              className="emotion-4 emotion-5"
            >
               Age 
              12
               
            </h2>
          </AgeHeading>
        </div>
      </ItemWrapper>
    </div>
  </HelloWrapper>
</Hello>
`;
